### DISCUSSION TITLE
2019/site-collection-app-catalog-vs-tenant-app-catalog/

### DISCUSSION THREAD

**Discussion by giscus** [mirontoli/chuvash-eu-comments#6] [General]
# 2019/site-collection-app-catalog-vs-tenant-app-catalog/

Site Collection App Catalogs are great for special cases (like developing apps or site unique apps), but using them on scale would be a mess.
I got a question: Why should we use the Tenant App Catalog at all when we could enable a Site Collection App Catalog on every teamsite? So the suggestion here is to install SharePoint Framework Packages on many Site Collection App Catalogs, instead of the Tenant App Catalog.

https://chuvash.eu/2019/site-collection-app-catalog-vs-tenant-app-catalog/

**Comment by mrstrong79** [comment]
Your point about Site Collection App Catalog apps being a security risk is incorrect. They are no more a security risk than apps deployed to the Tenant App Catalog. And as components run from SharePoint apps are run in the context of the current user there is no more security risk from SharePoint apps than any other components within SharePoint

**Reply by mirontoli** [reply]
If you control an app and its deployment, then there is no difference when it comes to security. My point is once you create Site App Catalog, any owner of the site can install a SharePoint app. If such an app has security issues, and you don't have a control of your site app catalogs then you have a problem. If someone wants to exploit this it is just to get owner access to a Site App Catalog and use it, this would be similar to CustomScript Risks.
